src/js/common/schemaform/helpers.js

```diff
@@ -318,7 +318,7 @@ export function updateRequiredFields(schema, uiSchema, formData) {
     const newRequired = Object.keys(schema.properties).reduce((requiredArray, nextProp) => {
       const field = uiSchema[nextProp];
       if (field && field['ui:required']) {
-        const isRequired = field['ui:required'](formData);
+        const isRequired = field['ui:required'](pageData, formData);
```

Comment:
If we keep the two variables, I think pageData should be first, since it should be the more common case. We shouldn't have a bunch of instances were data from one page is affecting fields in another.